File: RepoTransBench/target_projects/Python/C++/madisona_django-google-maps/public_tests/test_public_widget.cpp
#include <gtest/gtest.h>
#include <string>
#include <map>
#include <sstream>

std::string GoogleMapsAddressWidget_render_public(const std::string& name,
                                                 const std::string& value,
                                                 const std::map<std::string, std::string>& attrs) {
    std::ostringstream os;
    os << "<input";
    for (const auto& [k, v] : attrs)
        os << " " << k << "=\"" << v << "\"";
    os << " name=\"" << name << "\" type=\"text\"";
    if (!value.empty())
        os << " value=\"" << value << "\"";
    os << " />";
    os << "<div class=\"map_canvas_wrapper\">";
    os << "<div id=\"map_canvas\"></div></div>";
    return os.str();
}
std::string GoogleMapsAddressWidget_render_blank_public(const std::string& name,
                                                       const std::map<std::string, std::string>& attrs) {
    std::ostringstream os;
    os << "<input";
    for (const auto& [k, v] : attrs)
        os << " " << k << "=\"" << v << "\"";
    os << " name=\"" << name << "\" type=\"text\" />";
    os << "<div class=\"map_canvas_wrapper\">";
    os << "<div id=\"map_canvas\"></div></div>";
    return os.str();
}

std::string GoogleMapsApiJsUrl_public(const std::string& api_key) {
    return "https://maps.google.com/maps/api/js?key=" + api_key + "&libraries=places";
}

TEST(WidgetPublicTests, RenderReturnsCustomHtml) {
    std::string result = GoogleMapsAddressWidget_render_public("adam", "customvalue", {{"id", "unique"}, {"class", "css-test"}});
    std::string expected = "<input id=\"unique\" class=\"css-test\" name=\"adam\" type=\"text\" value=\"customvalue\" />";
    expected += "<div class=\"map_canvas_wrapper\">";
    expected += "<div id=\"map_canvas\"></div></div>";
    EXPECT_EQ(expected, result);
}

TEST(WidgetPublicTests, RenderReturnsBlankForValueWhenNonePublic) {
    std::string result = GoogleMapsAddressWidget_render_blank_public("foo", {{"style", "color:red;"}, {"data-bar", "hello"}});
    std::string expected = "<input style=\"color:red;\" data-bar=\"hello\" name=\"foo\" type=\"text\" />";
    expected += "<div class=\"map_canvas_wrapper\">";
    expected += "<div id=\"map_canvas\"></div></div>";
    EXPECT_EQ(expected, result);
}

TEST(WidgetPublicTests, MapsJsApiKeyDifferent) {
    std::string api_key = "OTHER_PUBLIC_KEY";
    std::string google_maps_js = GoogleMapsApiJsUrl_public(api_key);
    std::string widget_media_js1 = google_maps_js;
    EXPECT_EQ(google_maps_js, widget_media_js1);
}

File: RepoTransBench/target_projects/Python/Rust/madisona_django-google-maps/public_tests/test_public_widget.rs
#[cfg(test)]
mod public_widget {
    fn render_google_maps_address_widget(name: &str, value: Option<&str>, attrs: &[(&str, &str)]) -> String {
        let mut s = String::from("<input");
        for (k, v) in attrs.iter() {
            s.push(' ');
            s.push_str(k);
            s.push_str("=\"");
            s.push_str(v);
            s.push('"');
        }
        s.push_str(" name=\"");
        s.push_str(name);
        s.push_str("\" type=\"text\"");
        if let Some(val) = value {
            s.push_str(" value=\"");
            s.push_str(val);
            s.push('"');
        }
        s.push_str(" />");
        s.push_str("<div class=\"map_canvas_wrapper\">");
        s.push_str("<div id=\"map_canvas\"></div></div>");
        s
    }

    #[test]
    fn test_render_returns_custom_html() {
        let html = render_google_maps_address_widget(
            "adam",
            Some("customvalue"),
            &[("id", "unique"), ("class", "css-test")]
        );
        let expected = "<input id=\"unique\" class=\"css-test\" name=\"adam\" type=\"text\" value=\"customvalue\" />\
                        <div class=\"map_canvas_wrapper\"><div id=\"map_canvas\"></div></div>";
        assert_eq!(html, expected);
    }

    #[test]
    fn test_render_returns_blank_for_value_when_none_public() {
        let html = render_google_maps_address_widget(
            "foo",
            None,
            &[("style", "color:red;"), ("data-bar", "hello")]
        );
        let expected = "<input style=\"color:red;\" data-bar=\"hello\" name=\"foo\" type=\"text\" />\
                        <div class=\"map_canvas_wrapper\"><div id=\"map_canvas\"></div></div>";
        assert_eq!(html, expected);
    }

    #[test]
    fn test_maps_js_api_key_different() {
        let api_key = "___API_KEY___";
        let google_maps_js = format!("https://maps.google.com/maps/api/js?key={}&libraries=places", api_key);
        let simulated_media_js = vec!["foo.js".to_string(), google_maps_js.clone()];
        assert_eq!(google_maps_js, simulated_media_js[1]);
    }
}